### PR TITLE
[feat]: [CDS-58159]: add methods to fetch input yaml  env/ env-service override

### PR DESCRIPTION
**Pull request by tchaurasiya**
## Describe your changes

Adding methods to fetch runtime input for EnvServiceOverride and EnvironmentOverride
For both of the entities runtime input yaml will be same (for backward compatibility)
EnvironmentOverride runtime input yaml is same as Environment runtime input yaml

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48647)

### CODE REVIEW

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java [hunk 1]
```diff
@@ -348,4 +578,123 @@ private String createOverrideIdentifier(NGServiceOverridesEntity requestServiceO
...
+        new Criteria().orOperator(
+            Criteria.where(NGServiceOverridesEntity.NGServiceOverridesEntityKeys.projectIdentifier).exists(false),
+            Criteria.where(NGServiceOverridesEntity.NGServiceOverridesEntityKeys.projectIdentifier).is(null)),
+        additionalCriteria);
+  }
+
+  private Map<Scope, NGServiceOverridesEntity> getScopedEntities(
+      @NonNull String accountId, String orgId, String projectId, Criteria criteria) {
+    Map<Scope, NGServiceOverridesEntity> scopedEntities = new HashMap<>();
+    List<NGServiceOverridesEntity> projectScopedEntity =
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java:624]: we should check for project id being not empty

Comment (yogesh-chauhan) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java:624]: same for org

**Review by tchaurasiya — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48647)* status: 12 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 305 at r2](https://reviewable.io/reviews/harness/harness-core/48647#gh-1213122199:-NWrDmQcCrAel3W2hZSI:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6d4a2a3ea87775f4cc03264650f75f080c72e864/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L305)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

Lets use CDYamlUtils
</blockquote></details>

Done.
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 624 at r2](https://reviewable.io/reviews/harness/harness-core/48647#gh-1213120376:-NWrDnP73b-990S7K-EU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6d4a2a3ea87775f4cc03264650f75f080c72e864/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L624)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

same for org
</blockquote></details>

Done.
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 631 at r2](https://reviewable.io/reviews/harness/harness-core/48647#gh-1213115194:-NWrDnbe8Puq88Fn9rQz:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6d4a2a3ea87775f4cc03264650f75f080c72e864/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L631)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

can we put a warning if there are more than 1 overrides
</blockquote></details>

Done.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 4 of 6 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48647)* status: 12 file(s) to review. 1 of 2 approvals obtained.

**Comment by tchaurasiya** [comment]
trigger uts

**Comment by tchaurasiya** [comment]
trigger uts

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48647)* status: 13 file(s) to review. 0 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 3 of 6 files at r3, 3 of 3 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48647)* status: :shipit: complete! ✓

**Comment by tchaurasiya** [comment]
trigger review

**Pull request merged**